msal-client-credentials-flow/README.md

```diff
@@ -0,0 +1,163 @@
+```console
+$ curl http://localhost:8080/
+{
+  "@odata.context": "https://graph.microsoft.com/v1.0/$metadata#applications/$entity",
+  "id": "003c4673-d4a0-4eb6-8979-d6269e08b87e",
+  "deletedDateTime": null,
+  "appId": "c085fc22-2484-41c0-b795-2e940f51df64",
+  "applicationTemplateId": null,
+  "disabledByMicrosoftStatus": null,
+  "createdDateTime": "2021-12-17T16:45:43Z",
+  "displayName": "Node Web API",
+  "description": null,
+  "groupMembershipClaims": null,
+  "identifierUris": ["api://c085fc22-2484-41c0-b795-2e940f51df64"],
+  "isDeviceOnlyAuthSupported": null,
+  "isFallbackPublicClient": true,
+  "notes": null,
+  "publisherDomain": "contoso.onmicrosoft.com",
+  "serviceManagementReference": null,
+  "signInAudience": "AzureADMyOrg",
+  "tags": [],
+  "tokenEncryptionKeyId": null,
+  "defaultRedirectUri": null,
+  "certification": null,
+  "optionalClaims": null
+  …
+}
+```
+## Prerequisites
+
+- Azure Active Directory (Azure AD) tenant and the permissions or role required for managing app registrations in the tenant.
+- Node.js 16+
+
+## Setup
+
+### 1. Register the app
+
+First, complete the steps in [Quickstart: Register an application with the Microsoft identity platform](https://docs.microsoft.com/azure/active-directory/develop/quickstart-register-app) to register the sample API.
+
+Use these settings in your app registration.
+
+| App registration <br/> setting    | Value for this sample app                                                    | Notes                                                                                              |
+|:---------------------------------:|:-----------------------------------------------------------------------------|:---------------------------------------------------------------------------------------------------|
+| **Name**                          | `Node Web API`                                                               | Suggested value for this sample. <br/> You can change the app name at any time.                    |
+| **Supported account types**       | **Accounts in this organizational directory only (Single tenant)**           | Suggested value for this sample.                                                                   |
+| **Platform type**                 | _None_                                                                       | No redirect URI required; don't select a platform.                                                                    |
+| **Client secret**                 | _**Value** of the client secret (not its ID)_                                | :warning: Record this value immediately! <br/> It's shown only _once_ (when you create it).        |
+
+> :information_source: **Bold text** in the tables above matches (or is similar to) a UI element in the Azure portal, while `code formatting` indicates a value you enter into a text box in the Azure portal.
+
+### 2. Update code sample with app registration values
```

Comment:
Customer: "Which file do I update?"

Always be explicit. Never leave anything up for inference. For example, always specify the file name and path _even when there's only ONE file in the project_.

Never require someone to infer something that might seem obvious to you - to them, it's not. They'll get it wrong or won't even try, and in comes the GitHub issues. Guaranteed. ☹️